cardano-node/Cardano/Node/CLI.hs

```diff
@@ -93,28 +110,43 @@ data SomeProtocol where
   SomeProtocol :: (RunDemo blk, TraceConstraints blk)
                => Consensus.Protocol blk -> SomeProtocol
 
-fromProtocol :: Protocol -> IO SomeProtocol
-fromProtocol BFT =
+fromProtocol :: CardanoConfiguration -> Protocol -> IO SomeProtocol
+fromProtocol _ BFT =
     case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
     p = ProtocolMockBFT defaultSecurityParam
-fromProtocol Praos =
+fromProtocol _ Praos =
     case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
     p = ProtocolMockPraos defaultDemoPraosParams
-fromProtocol MockPBFT =
+fromProtocol _ MockPBFT =
     case Consensus.runProtocol p of
       Dict -> return $ SomeProtocol p
   where
     p = ProtocolMockPBFT defaultDemoPBftParams
-fromProtocol RealPBFT =
+fromProtocol cc@CardanoConfiguration{..} RealPBFT = do
+    let Genesis{..} = coGenesis ccCore
+        genHash = either (throw . ConfigurationError) id $ -- XXX: need a proper error tag
+                  decodeAbstractHash geGenesisHash
+        cvtRNM :: RequireNetworkMagic -> RequiresNetworkMagic
+        cvtRNM NoRequireNetworkMagic = RequiresNoMagic
+        cvtRNM RequireNetworkMagic   = RequiresMagic
+        StaticKeyMaterial{..} = coStaticKeyMaterial ccCore
+        kmoDeserialiseDelegateKey = Signing.SigningKey . snd . either (error . show) id . deserialiseFromBytes Signing.fromCBORXPrv
+    genesisConfig <- either (error . show) id <$> -- XXX: need a proper error tag
+      runExceptT (Genesis.mkConfigFromFile (cvtRNM $ coRequiresNetworkMagic ccCore) geSrc genHash)
+
+    sk  <- kmoDeserialiseDelegateKey <$> LB.readFile skmSigningKeyFile
+    dlg <- either (error . show) id . CanonicalJSON.canonicalDecPre <$> LB.readFile skmDlgCertFile
+
+    let skm = StaticKeyMaterialByronPbft sk dlg
+        p   = ProtocolRealPBFT defaultDemoPBftParams genesisConfig skm
+
     case Consensus.runProtocol p of
-      Dict -> return $ SomeProtocol p
-  where
-    p = ProtocolRealPBFT defaultDemoPBftParams genesisConfig
-    genesisConfig = Dummy.dummyConfig
+      Dict -> do
+        return . SomeProtocol $ p
```

Comment:
If you write it this way `return $ SomeProtocol p` it will be one character less.  :thinking: I just wrote a whole sentence to reduce a single character :D.

Reply:
Interesting -- this again refers to an outdated commit -- this was recently shortened.